lib/open-search-client.ts

```diff
@@ -380,3 +384,68 @@ export const deleteIndex = async (
 ): Promise<void> => {
   await gotEs(prefixUrl).delete(`${index}`);
 };
+
+const scrollResponseSchema = z.object({
+  _scroll_id: z.string(),
+  hits: z.object({
+    hits: z.array(z.unknown()),
+  }),
+});
+
+/**
+ * Iterate over ElasticSearch Scroll API hits, applying `f` to each hit.
+ */
+export const scrollMap = async (params: {
+  esUrl: string,
+  index: string,
+  handler: (i: unknown) => Promise<void>,
+  pMapOptions?: pMap.Options,
+  scrollOptions?: OpenScrollOptions
+}): Promise<void> => {
+  const {
+    esUrl,
+    index,
+    handler: f,
+    pMapOptions,
+    scrollOptions,
+  } = params;
+
+  const rawOpenScrollResponse = await openScroll(esUrl, index, scrollOptions);
+
+  const openScrollResponse = scrollResponseSchema.parse(rawOpenScrollResponse);
+
+  const scrollId = openScrollResponse._scroll_id;
+
+  let { hits } = openScrollResponse.hits;
+
+  let ctr = 0;
+
+  try {
+    /* eslint-disable no-await-in-loop */
+    while (hits.length > 0 && ctr < 2) {
+      await pMap(hits, f, pMapOptions);
+
+      const nextScrollResponse = scrollResponseSchema.parse(
+        await nextScroll(esUrl, scrollId)
+      );
+
+      hits = nextScrollResponse.hits.hits;
+
+      ctr += 1;
```

Comment:
So at most we only loop through hits twice? What if we have more searchable documents than that?

Reply:
That was for testing and should have been removed. Fixed in 0cb0177